### DISCUSSION TITLE
Better command line arguments management with Click

### DISCUSSION THREAD

**Discussion by mavwolverine** [piccolo-orm/piccolo#507] [Ideas]
Currently the code is relying on sys.argv.

https://click.palletsprojects.com/en/8.1.x/ is a great library for arguments passing to command line.

Might be a good improvement

**Comment by dantownsend** [comment]
We use this library currently for most of our CLI: https://github.com/piccolo-orm/targ

You're right, we do use `sys.argv` in a few places.

The reason we made something custom rather than using click, is it gives us a bit more control over the look of our CLI. And it makes [registering custom CLI commands](https://piccolo-orm.readthedocs.io/en/latest/piccolo/projects_and_apps/piccolo_apps.html#commands) really easy.
